packages/headless/src/modules/super-admin/service/index.ts

```diff
@@ -1,59 +1,295 @@
+      email: body.email,
+    });
+    if (!userData) {
+      return errorResponse(
+        'Invalid credentials!',
+        null,
+        HttpStatus.BAD_REQUEST,
+      );
+    }
+
+    const matchPassword = await bcrypt.compare(
+      body.password,
+      userData.password,
+    );
+    if (!matchPassword) {
+      return errorResponse(
+        'Invalid credentials!',
+        null,
+        HttpStatus.BAD_REQUEST,
+      );
+    }
+
+    if (userData.isMfaEnabled) {
+      userData.password = body.password;
+      const res = await this.handleMfaLogin(userData);
+      return res;
+    }
+
+    const payload: AdminJwtPayload = {
+      id: userData.id,
+      username: userData.firstName + ' ' + userData.lastName,
+      logInTime: Date.now(),
+      role: userData.role,
+    };
+    const token = this.jwtService.sign(payload);
+    return successResponse(SuperAdminLoginRes, { token });
+  }
+
+  async verifyMfaLoginOtp(
+    body: MfaVerifyOtpDto,
+  ): Promise<IServiceResponse<SuperAdminLoginRes>> {
+    if (!body.email && !body.phone) {
+      return errorResponse(
+        'Require email or phone',
+        null,
+        HttpStatus.BAD_REQUEST,
+      );
+    }
+    let query: Record<string, any> = {
+      isVerified: false,
+      id: body.id,
+      otp: body.otp,
+      otpExpireTime: { $gt: Date.now() },
+    };
+
+    if ((body.email && !body.phone) || (body.email && body.phone)) {
+      query.email = body.email;
+    } else {
+      query.phone = body.phone;
+    }
+
+    const verifiedData = await this.superAdminRepository.verifyOtp(query);
+    if (verifiedData) {
+      await this.superAdminRepository.deleteOtp({
+        ...query,
+        isVerified: true,
+      });
+      const userData = await this.superAdminRepository.findOne({
+        id: verifiedData.userId,
+      });
+      const payload: AdminJwtPayload = {
+        id: userData.id,
+        username: userData.firstName + ' ' + userData.lastName,
+        logInTime: Date.now(),
+        role: userData.role,
+      };
+      const token = this.jwtService.sign(payload);
+      return successResponse(SuperAdminLoginRes, { token });
+    } else {
+      return errorResponse('OTP is not verified', null, HttpStatus.CONFLICT);
+    }
+  }
+
+  //--- Multi factor auth functionalities---
+
+  async sendMfaOtp(
+    userId: string,
+    body: MfaOtpDto,
+  ): Promise<IServiceResponse<OtpResponseDto>> {
+    if (!body.email && !body.phone) {
+      return errorResponse(
+        'Require email or phone',
+        null,
+        HttpStatus.BAD_REQUEST,
+      );
+    }
+
+    const isPasswordMatched = await this.superAdminHelperService.checkPassword(
+      body,
+      userId,
+    );
+    if (!isPasswordMatched) {
+      return errorResponse(
+        'Require valid password',
+        null,
+        HttpStatus.BAD_REQUEST,
+      );
+    }
+    let isEmail = false;
+    if ((body.email && !body.phone) || (body.email && body.phone)) {
```

Comment:
this condition is simply equivalent to `if (body.email) `. the other condition is just useless. correct me if i am wrong